esquery.js

```diff
@@ -298,10 +298,15 @@ function generateMatcher(selector) {
 
         case 'class': {
 
-            const name = selector.name.toLowerCase();
             return (node, ancestry, options) => {
-
+                
+                if (options && options.matchClass) {
+                    return options.matchClass(selector.name, node, ancestry);
+                }
+                
                 if (options && options.nodeTypeKey) return false;
+                
+                const name = selector.name.toLowerCase();
```

Comment:
Nit: Reading the code it looks like the selector name doesn't change. Moving that call back into the closure leads to `.toLowerCase()` being called repeatedly with the same value. That's work that can be avoided and I guess that was the reason it was hoisted outside of the function closure here.

Reply:
Probably left from the multiple rounds of refactoring. 👍